### PR TITLE
unorganized commit | List Structure , collapsible menu & unit test

### PR DESCRIPTION
**Pull request by ashik-75**
# :white_check_mark: UI 

- [ ] Ensure there are no ESLint errors
- [ ] Identify and resolve any new warnings in the console
- [ ] Identify and resolve any new errors in the console 
- [x] If there are modifications to configurations such as `vite.config.ts`, `tsconfig.json` etc., ensure to include a description
- [x] Ensure all the tests are passing

# :white_large_square: Automation Scripts

- [ ] Ensure there are no ESLint errors
- [ ] Identify and resolve any new warnings in the console
- [ ] Identify and resolve any new errors in the console 
- [ ] If there are modifications to configurations such as `vite.config.ts`, `tsconfig.json` etc., ensure to include a description
- [ ] Ensure all the tests are passing

### CODE REVIEW

File: ui/vite.config.ts [hunk 1]
```diff
@@ -1,8 +1,17 @@
...
+/// <reference types="vite/client" />
+
 import { defineConfig } from "vite";
 import react from "@vitejs/plugin-react";
 import viteTsconfigPaths from "vite-tsconfig-paths";
 
 // https://vitejs.dev/config/
 export default defineConfig({
 	plugins: [react(), viteTsconfigPaths()],
+	test: {
```
Comment (lifeparticle) [inline ui/vite.config.ts:11]: Add the doc link

File: ui/src/test/setup.ts [hunk 2]
```diff
@@ -0,0 +1 @@
+import "@testing-library/jest-dom";
```
Comment (lifeparticle) [inline ui/src/test/setup.ts:1]: Maybe rv? We already have a setupTests.ts file

File: ui/src/pages/list/components/types.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+interface ListProps<T> {
```
Comment (lifeparticle) [inline ui/src/pages/list/components/types.ts:1]: Rm List folder concept

File: ui/src/pages/list/components/List.tsx [hunk 4]
```diff
@@ -0,0 +1,64 @@
+import React, { ChangeEvent, useState } from "react";
+import { useSearchParams } from "react-router-dom";
+import { Input } from "antd";
+import style from "./resource.module.scss";
+
+import { ListProps } from "./types";
```
Comment (lifeparticle) [inline ui/src/pages/list/components/List.tsx:6]: Double check

File: ui/src/pages/list/Youtube/index.tsx [hunk 5]
```diff
@@ -0,0 +1,17 @@
+import List from "components/Hoc/List/List";
+import { programmingChannels } from "./data";
+import Resource from "components/General/ListItems/Resource/Resource";
+
+const YouTubeList: React.FC = () => {
+	return (
+		<>
```
Comment (lifeparticle) [inline ui/src/pages/list/Youtube/index.tsx:7]: Rem extra fragment

File: ui/src/pages/list/Youtube/data.ts [hunk 6]
```diff
@@ -0,0 +1,74 @@
+const programmingChannels = [
+	{
+		name: "Traversy Media",
+		category: "Web Development",
+		subCategory: ["HTML/CSS", "JavaScript"],
+		url: "https://www.youtube.com/user/TechGuyWeb",
```
Comment (lifeparticle) [inline ui/src/pages/list/Youtube/data.ts:6]: move into public folder like news,

list folder -> youtube.json, github.json, icon.json

File: ui/src/pages/list/Icons/index.tsx [hunk 7]
```diff
@@ -0,0 +1,17 @@
+import List from "components/Hoc/List/List";
+import { iconsList } from "./data";
+import Resource from "components/General/ListItems/Resource/Resource";
+
+const IconList: React.FC = () => {
+	return (
+		<>
```
Comment (lifeparticle) [inline ui/src/pages/list/Icons/index.tsx:7]: Rm fragment

File: ui/src/pages/list/Icons/data.ts [hunk 8]
```diff
@@ -0,0 +1,49 @@
+const iconsList = [
+	{
+		name: "Tabler Icons",
+		category: "Frontend",
+		subCategory: ["React", "Frontend"],
```
Comment (lifeparticle) [inline ui/src/pages/list/Icons/data.ts:5]: move it json file

File: ui/src/pages/list/Github/data.ts [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+	},
+	{
+		name: "Vue.js",
+		category: "Frontend",
+		subCategory: ["Vue.js", "Frontend"],
+		url: "https://github.com/vuejs/vue",
+		socials: [{ url: "https://vuejs.org/", name: "website" }],
+	},
+	{
+		name: "Node.js",
```
Comment (lifeparticle) [inline ui/src/pages/list/Github/data.ts:17]: mv it json file

File: ui/src/pages/YouTubeList/YouTubeList.tsx [hunk 10]
```diff
@@ -1,10 +1,15 @@
+import List from "components/Hoc/List/List";
 import { programmingChannels } from "./data";
-import ResourceList from "components/Hoc/List";
+import Resource from "components/General/ListItems/Resource/Resource";
 
 const YouTubeList: React.FC = () => {
 	return (
 		<>
-			<ResourceList listData={programmingChannels} />
```
Comment (lifeparticle) [inline ui/src/pages/YouTubeList/YouTubeList.tsx:7]: rm fragment

File: ui/src/pages/News/index.tsx [hunk 11]
```diff
@@ -1,21 +1,16 @@
...
-const News = () => {
-	const location = useLocation();
-
-	const endpoint =
-		location.search?.length > 3 ? location.search : "?q=javascript";
-
+const NewsPage = () => {
 	const { data, isLoading, isError } = useQuery<APIResponse>({
-		queryKey: [endpoint],
+		queryKey: ["news"],
```
Comment (lifeparticle) [inline ui/src/pages/News/index.tsx:13]: No magic string

File: ui/src/pages/News/components/types.ts [hunk 12]
```diff
@@ -0,0 +1,10 @@
+interface ListProps<T> {
+	items: T[];
+	resourceName: string;
+	itemComponent: React.FC<{
+		resource: T;
+		handleOnClick: (url: string) => void;
+	}>;
```
Comment (lifeparticle) [inline ui/src/pages/News/components/types.ts:7]: rm

File: ui/src/pages/News/components/List.tsx [hunk 13]
```diff
@@ -0,0 +1,54 @@
+import React, { ChangeEvent, useState } from "react";
+import { useSearchParams } from "react-router-dom";
+import { Input } from "antd";
+import style from "./resource.module.scss";
+
+import { ListProps } from "./types";
+import { Article } from "../types.ts/types";
+
+const List: React.FC<ListProps<Article>> = ({
```
Comment (lifeparticle) [inline ui/src/pages/News/components/List.tsx:9]: rm file

File: ui/src/pages/IconList/index.tsx [hunk 14]
```diff
@@ -1 +1,17 @@
-export { default } from "./IconList";
+import List from "components/Hoc/List/List";
+import { iconsList } from "./data";
+import Resource from "components/General/ListItems/Resource/Resource";
+
+const IconList: React.FC = () => {
+	return (
+		<>
```
Comment (lifeparticle) [inline ui/src/pages/IconList/index.tsx:7]: rm fragment

File: ui/src/pages/IconList/data.ts [hunk 15]
```diff
@@ -0,0 +1,49 @@
...
+		url: "https://tabler-icons.io/",
+		socials: [
+			{ url: "https://github.com/tabler/tabler-icons", name: "github" },
+		],
+	},
+	{
+		name: "Tabler Icons",
+		category: "Frontend",
+		subCategory: ["React", "Frontend"],
+		url: "https://tabler-icons.io/",
```
Comment (lifeparticle) [inline ui/src/pages/IconList/data.ts:15]: mv it json file

File: ui/src/pages/GithubList/index.tsx [hunk 16]
```diff
@@ -1,10 +1,15 @@
-import ResourceList from "components/Hoc/List";
 import { repositories } from "./data";
+import List from "components/Hoc/List/List";
+import Resource from "components/General/ListItems/Resource/Resource";
 
 function GithubList() {
 	return (
 		<div>
-			<ResourceList listData={repositories} />
```
Comment (lifeparticle) [inline ui/src/pages/GithubList/index.tsx:7]: rm div

File: ui/src/main.tsx [hunk 17]
```diff
@@ -18,11 +18,9 @@ const client = new QueryClient({
 });
 
 root.render(
-	<React.StrictMode>
-		<QueryClientProvider client={client}>
-			<BrowserRouter>
```
Comment (lifeparticle) [inline ui/src/main.tsx:23]: Add back strict mode

File: ui/src/components/Hoc/List/List.tsx [hunk 18]
```diff
@@ -42,7 +53,7 @@ const List: React.FC<ListProps> = ({
 				{filteredList.map((item, i) => (
 					<ItemComponent
 						key={i}
-						{...{ [resourceName]: item }}
```
Comment (lifeparticle) [inline ui/src/components/Hoc/List/List.tsx:45]: Keep the old code

File: ui/src/components/General/ListItems/Resource/Resource.tsx [hunk 19]
```diff
@@ -8,12 +8,18 @@ import {
...
+		url: string;
+	};
+	handleOnClick: (url: string) => void;
 }
 
-const Resource: React.FC<ResourceProps> = ({ resource }) => {
+const Resource: React.FC<ResourceProps> = ({ resource, handleOnClick }) => {
 	const { name, category, subCategory, url } = resource;
-
+	console.log({ resource });
```
Comment (lifeparticle) [inline ui/src/components/General/ListItems/Resource/Resource.tsx:22]: rm console

File: ui/vite.config.ts (hunk 1 above)
Comment (ashik-75) [reply ui/vite.config.ts:11]: https://github.com/vitest-dev/vitest/tree/main/examples/react-testing-lib

File: ui/tsconfig.json [hunk 20]
```diff
@@ -22,6 +22,6 @@
 		"noUnusedParameters": true,
 		"noFallthroughCasesInSwitch": true
 	},
-	"include": ["src", "src/**/*.spec.ts"],
+	"include": ["src", "public/list/data.ts"],
```
Comment (lifeparticle) [inline ui/tsconfig.json:25]: rm extra path

File: ui/src/pages/IconList/index.tsx [hunk 21]
```diff
@@ -1 +1,39 @@
...
+		queryFn: () => {
+			return getData(URL);
+		},
+	});
+	return (
+		<div>
+			{isLoading ? (
+				<Spin
+					indicator={
+						<LoadingOutlined style={{ fontSize: 34 }} spin />
```
Comment (lifeparticle) [inline ui/src/pages/IconList/index.tsx:23]: pass loading and error as a props

File: ui/src/components/General/ListItems/News/News.tsx [hunk 22]
```diff
@@ -1,47 +1,24 @@
...
-				<SettingOutlined key="setting" />,
-				<EditOutlined key="edit" />,
-				<EllipsisOutlined key="ellipsis" />,
-			]}
-			key={name}
-			hoverable
-			onClick={() => handleOnClick(url)}
-		>
-			<Skeleton loading={false} avatar active>
-				<Meta
```
Comment (lifeparticle) [inline ui/src/components/General/ListItems/News/News.tsx:30]: bring back skelaton

**Pull request merged**